app/questionnaire/questionnaire_manager.py

```diff
@@ -110,9 +110,13 @@ def _conditional_display(self, item):
 
             if hasattr(item.schema_item, 'skip_condition') and item.schema_item.skip_condition:
                 rule = item.schema_item.skip_condition.as_dict()
-                answer = get_answers(current_user).get(rule['when']['id'])
 
-                item.skipped = evaluate_rule(rule, answer)
+                if 'id' in rule['when'] and rule['when']['id']:
+                    answer = get_answers(current_user).get(rule['when']['id'])
+                    item.skipped = evaluate_rule(rule, answer)
+
+                if 'meta' in rule['when'] and rule['when']['meta']:
+                    item.skipped = evaluate_metadata_rule(rule, get_metadata(current_user))
```

Comment:
This logic looks the same as used in the `evaluate_goto` method in Navigator, any way we could rationalise this?